### ISSUE TITLE
v3-preview: Could not load file or assembly 'netstandard, Version=2.1.0.0

### ISSUE THREAD

**Issue by DrBarnabus** [Azure/azure-functions-host#5224]
I am attempting to update our Azure Functions Projects to the v3-preview for netcoreapp3.0 support. However, I have encountered an issue and as suggested in https://dev.to/azure/develop-azure-functions-using-net-core-3-0-gcm I am logging it here.

I keep getting the following error while trying to build the project:

```
Microsoft.NET.Sdk.Functions.Generator.dll "C:\dev\repos\Eruza\CS.Eruza\src\CS.Eruza.Functions.Administrator\bin\Debug\netcoreapp3.0\bin\CS.Eruza.Functions.Administrator.dll " "C:\dev\repos\Eruza\CS.Eruza\src\CS.Eruza.Functions.Administrator\bin\Debug\netcoreapp3.0\ " "False "
6>    C:\Users\<UserName>\.nuget\packages\microsoft.net.sdk.functions\1.0.30-beta2\build\Microsoft.NET.Sdk.Functions.Build.targets(41,5): error : System.IO.FileNotFoundException: Could not load file or assembly 'netstandard, Version=2.1.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. The system cannot find the file specified.
6>    C:\Users\<UserName>\.nuget\packages\microsoft.net.sdk.functions\1.0.30-beta2\build\Microsoft.NET.Sdk.Functions.Build.targets(41,5): error : File name: 'netstandard, Version=2.1.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'
6>    C:\Users\<UserName>\.nuget\packages\microsoft.net.sdk.functions\1.0.30-beta2\build\Microsoft.NET.Sdk.Functions.Build.targets(41,5): error :    at System.Reflection.RuntimeAssembly.GetExportedTypes(RuntimeAssembly assembly, ObjectHandleOnStack retTypes)
6>    C:\Users\<UserName>\.nuget\packages\microsoft.net.sdk.functions\1.0.30-beta2\build\Microsoft.NET.Sdk.Functions.Build.targets(41,5): error :    at System.Reflection.RuntimeAssembly.GetExportedTypes()
6>    C:\Users\<UserName>\.nuget\packages\microsoft.net.sdk.functions\1.0.30-beta2\build\Microsoft.NET.Sdk.Functions.Build.targets(41,5): error :    at MakeFunctionJson.FunctionJsonConverter.TryGenerateFunctionJsons()
6>    C:\Users\<UserName>\.nuget\packages\microsoft.net.sdk.functions\1.0.30-beta2\build\Microsoft.NET.Sdk.Functions.Build.targets(41,5): error :    at MakeFunctionJson.FunctionJsonConverter.TryRun()
6>    C:\Users\<UserName>\.nuget\packages\microsoft.net.sdk.functions\1.0.30-beta2\build\Microsoft.NET.Sdk.Functions.Build.targets(41,5): error : 
6>    C:\Users\<UserName>\.nuget\packages\microsoft.net.sdk.functions\1.0.30-beta2\build\Microsoft.NET.Sdk.Functions.Build.targets(41,5): error : 
6>    C:\Users\<UserName>\.nuget\packages\microsoft.net.sdk.functions\1.0.30-beta2\build\Microsoft.NET.Sdk.Functions.Build.targets(41,5): error : Error generating functions metadata
6>    C:\Users\<UserName>\.nuget\packages\microsoft.net.sdk.functions\1.0.30-beta2\build\Microsoft.NET.Sdk.Functions.Build.targets(41,5): error :
6>    C:\Users\<UserName>\.nuget\packages\microsoft.net.sdk.functions\1.0.30-beta2\build\Microsoft.NET.Sdk.Functions.Build.targets(41,5): error : Metadata generation failed.
6>  Done executing task "GenerateFunctions" -- FAILED.
```

My understanding was that v3-preview should support netcoreapp3.0 and thus also support netstandard2.1 however the error message seems to suggest otherwise.

Is this something that I can fix by changing some properties, or could it be a package dependency that is causing the issue?

Any help would be appreciated.

**Comment by bbougot** [comment]
I'm getting the exact same issue, after having follows **very carefully** this [topic](https://dev.to/azure/develop-azure-functions-using-net-core-3-0-gcm).

I've run some tests, and it seems adding EF Core 3.0 and a DBContext inside my function project breaks the thing.

**Comment by bbougot** [comment]
Main issue is tracked here: https://github.com/Azure/azure-functions-core-tools/issues/1642

**Comment by DrBarnabus** [comment]
I do have EF Core installed into my functions project, but my DBContext's are defined in a netstandard2.1 library.

From the little bit of testing I have done, just referencing something that uses Entity Framework causes the issue regardless of if you define a DbContext in that project.

**Comment by nbalu** [comment]
Ok, I've solved my issue:

It relates to the following pattern:
I've used Microsoft.Azure.Functions.Extensions, but left my generated function as is and thought that the injection will be happening on function level.

Which is not the case, I had to remove the static modifiers pass the injected variables through ctor and then the functions could be generated.

My bad. It works with EF Core 3 and in Ctor I'm able to pass the DBContext instance.
